src/ResourceManager/Search/Commands.Management.Search/SearchService/NewSearchServiceAdminKeyCommand.cs

```diff
@@ -0,0 +1,101 @@
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+using Microsoft.Azure.Management.Search.Models;
+using System;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.Management.Search.SearchService
+{
+    [Cmdlet(VerbsCommon.New, SearchServiceAdminKeyNounStr, SupportsShouldProcess = true), OutputType(typeof(PSSearchAdminKey))]
+    public class NewSearchServiceAdminKeyCommand : SearchServiceBaseCmdlet
+    {
+        [Parameter(
+           Position = 0,
+           Mandatory = true,
+           ValueFromPipeline = true,
+           ParameterSetName = ParentObjectParameterSetName,
+           HelpMessage = InputObjectHelpMessage)]
+        [ValidateNotNullOrEmpty]
+        public PSSearchService ParentObject { get; set; }
+
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            ParameterSetName = ParentResourceIdParameterSetName,
+            HelpMessage = ResourceIdHelpMessage)]
+        [ValidateNotNullOrEmpty]
+        public string ParentResourceId { get; set; }
+
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            ParameterSetName = ResourceNameParameterSetName,
+            HelpMessage = ResourceGroupHelpMessage)]
+        [ResourceGroupCompleter()]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Position = 1,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            ParameterSetName = ResourceNameParameterSetName,
+            HelpMessage = ResourceNameHelpMessage)]
+        [ValidateNotNullOrEmpty]
+        public string ServiceName { get; set; }
+
+        [Parameter(
+           Mandatory = true,
+           ValueFromPipelineByPropertyName = true,
+           HelpMessage = KeyKindHelpMessage)]
+        [ValidateNotNullOrEmpty]
+        public AdminKeyKind KeyKind { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = ForceHelpMessage)]
+        public SwitchParameter Force { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            if (ParameterSetName.Equals(ParentObjectParameterSetName, StringComparison.InvariantCulture))
+            {
+                ResourceGroupName = ParentObject.ResourceGroupName;
+                ServiceName = ParentObject.Name;
+            }
+            else if (ParameterSetName.Equals(ParentResourceIdParameterSetName, StringComparison.InvariantCulture))
+            {
+                var id = new ResourceIdentifier(ParentResourceId);
+                ResourceGroupName = id.ResourceGroupName;
+                ServiceName = id.ResourceName;
+            }
+
+            ConfirmAction(Force.IsPresent,
```

Comment:
You check for Force.IsPresent, but you don't have a force parameter

Reply:
You mean in help file? Added there.